### ISSUE TITLE
(URGENT) `ftp://www.3gpp.org` no longer has bibliographic database

### ISSUE THREAD

**Issue by ronaldtse** [relaton/relaton-3gpp#18]
From @andrew2net :

> looks like 3GPP doesn't have `ftp://www.3gpp.org/Information/Databases/Spec_Status/` folder on their FTP server anymore. Could you ask them how to get updates now?

For a very long time, 3GPP provided bibliographic entries through `ftp://www.3gpp.org/` in form of the `status_smg_3GPP.mdb` Access database file.

Unfortunately, last week the 3GPP FTP site was updated, and that file no longer exists. This means that the IETF BibXML service (and hence RFCs and Internet-Drafts) would no longer be able to cite 3GPP publications moving forward.

**Comment by ronaldtse** [comment]
I have contacted @haudiobe to see if he would be able to help contact 3GPP. Thank you @haudiobe in advance!

**Comment by andrew2net** [comment]
@ronaldtse the new data source has duplicates like:
```
03.20ext;Security-related fetwork functions (Ext);https://www.3gpp.org/ftp/Specs/archive/03_series/03.20ext/0320ext-300.zip;3.0.0;Jan  1 1995 12:00AM;1;Nguyen Ngoc;Sebastien;Orange;S3;R6, R6, CP;Phase 1 extension;GSM_PH1;;;;;SA#25;Oct  1 1994 12:00AM;
03.20ext;Security-related fetwork functions (Ext);https://www.3gpp.org/ftp/Specs/archive/03_series/03.20ext/0320ext-300.zip;3.0.0;Sep  1 2003 12:00AM;1;Nguyen Ngoc;Sebastien;Orange;S3;R6, R6, CP;Phase 1 extension;GSM_PH1;;;;;SA#25;Oct  1 1994 12:00AM;
```
the only difference is date, `Jan  1 1995` vs `Sep  1 2003`. Should we rewrite the oldest date with the most recent? Or should we add a year to ID?

**Comment by ronaldtse** [comment]
@andrew2net are these the only problematic ones? Can you list them all out so we can ask 3GPP all at once?

**Comment by andrew2net** [comment]
@ronaldtse there are 1624 duplication groups. They are grouped by "Spec number" and "Version". Each group has from 2 to 5 duplications. See all list in the attachment:
[3gpp_dups.txt](https://github.com/relaton/relaton-3gpp/files/14104799/3gpp_dups.txt)

**Comment by ronaldtse** [comment]
Here are the two questions asked to 3GPP:

1. Some CSV lines have identical content but different dates
2. Some CSV lines have identical content but different contributors

Answer from Dongwook Kim of 3GPP:

1. The difference is that whether the version has a corresponding transposed standards by the organizing partner SDOs or not. The version in 1995 does not have a transposed version, whereas the version in 2003 does.
2. Yes your understanding is correct. There are multiple rapporteurs for 37.885.

**Comment by ronaldtse** [comment]
Question:

Regarding question 1: so this is the same document, that was originally published in 1995, and then it was republished with a partner SDO in 2005? Or can there be a flag to explicitly say that it was re-published at the partner SDO?

Answer from Dongwook Kim of 3GPP:

It is possible to create a flag, but I would prefer (for simplicity reasons) to have another column name “ETSI Transposition” that states the ETSI standard to which the 3GPP specification was transposed to.
For the example below, it would be NULL for 1995, and something like “ETSI TS 103 020ext…”.

**Comment by ronaldtse** [comment]
Answer from Dongwook Kim:

Just to confirm, you are OK if it can be distinguished by “NULL” (no transposition) or any other values (yes transposition), right?

If so, I will update the excel accordingly from this March plenary 😊

-----

@andrew2net do we have sufficient information? Thanks!

**Comment by ronaldtse** [comment]
Thank you @haudiobe  - I think we have everything sorted out with the help of Dongwook. Thanks!

**Comment by ronaldtse** [comment]
The data is at:
* https://www.3gpp.org/ftp/Information/Databases/3GPPBibliography_20240122.csv

Future:
* https://www.3gpp.org/ftp/Information/Databases/3GPPBibliography.csv

**Comment by andrew2net** [comment]
@ronaldtse so what does the "transposed" date mean in the Relaton context? Should we save the date as a specific type?
We have following date types:
```xml
  <define name="BibliographicDateType">
    <choice>
      <value>published</value>
      <value>accessed</value>
      <value>created</value>
      <value>implemented</value>
      <value>obsoleted</value>
      <value>confirmed</value>
      <value>updated</value>
      <value>corrected</value>
      <value>issued</value>
      <value>transmitted</value>
      <value>copied</value>
      <value>unchanged</value>
      <value>circulated</value>
      <value>adapted</value>
      <value>vote-started</value>
      <value>vote-ended</value>
      <value>announced</value>
      <value>stable-until</value>
    </choice>
  </define>
```
Which one should we use?

**Comment by ronaldtse** [comment]
I'm not sure if we have any corresponding relation. What they mean by "transpose" is that the document has been sent to be published in a new place.

So  instead of a date, it should be a [standard relationship](https://www.relaton.org/model/relations/) of:
* `adoptedAs:equivalent`

**Comment by andrew2net** [comment]
@ronaldtse we don't have any "transpose" flag in the [dataset](https://www.3gpp.org/ftp/Information/Databases/3GPPBibliography.csv). Can we infer that the earlier published document is a main publication and all others are `adoptedAs` relations?

**Comment by ronaldtse** [comment]
Yes. That is the correct understanding. The first item that is published is the real one. The subsequent items that are “transposed” means they have been published at other venues with the same content and unmodified, so these are adoptions.

**Comment by andrew2net** [comment]
I found duplication with missing a link and a date in one of 2 rows. It looks like we should drop the row with missing data. I just put the example here for information. 
```csv
22.135,Multicall; Service description; Stage 1,,4.2.0,,1,Miettinen,Natalia,NOKIA UK R&D Ltd,S1, ,Release 4,GSM_Release_4,3G_Rel-4,SA-#11,SA-#11,SA-#11,SA#65,Aug  1 1998 12:00AM,Jun 21 2001 12:00AM
22.135,Multicall; Service description; Stage 1,https://www.3gpp.org/ftp/Specs/archive/22_series/22.135/22135-420.zip,4.2.0,Dec 18 2002 12:00AM,1,Miettinen,Natalia,NOKIA UK R&D Ltd,S1, ,Release 4,GSM_Release_4,3G_Rel-4,SA-#11,SA-#11,SA-#11,SA#65,Aug  1 1998 12:00AM,Jun 21 2001 12:00AM
```

**Comment by andrew2net** [comment]
@ronaldtse the new dataset doesn't have `editorialgroup`, but our datamodel requires it.
@opoudjis could we make `editorialgroup` optional in the `BibDataExtensionType` definition in `relaton-3gpp.rng`?

**Comment by ronaldtse** [comment]
> @ronaldtse the new dataset doesn't have `editorialgroup`, but our datamodel requires it. @opoudjis could we make `editorialgroup` optional in the `BibDataExtensionType` definition in `relaton-3gpp.rng`?

Yes, thanks @andrew2net .

**Comment by andrew2net** [comment]
Parser for the new datase is implemented in v1.19.1
